cluster-autoscaler/core/scale_down.go

```diff
@@ -876,7 +876,7 @@ func (sd *ScaleDown) TryToScaleDown(
 	emptyNodes := sd.getEmptyNodes(candidateNames, sd.context.MaxEmptyBulkDelete, scaleDownResourcesLeft)
 	if len(emptyNodes) > 0 {
 		nodeDeletionStart := time.Now()
-		deletedNodes, err := sd.scheduleDeleteEmptyNodes(emptyNodes, sd.context.ClientSet, sd.context.Recorder, readinessMap, candidateNodeGroups)
+		deletedNodes, err := sd.scheduleDeleteEmptyNodes(emptyNodes, sd.context.DaemonSetEvictionForEmptyNodes, sd.context.ClientSet, sd.context.Recorder, readinessMap, candidateNodeGroups)
```

Comment:
I would suggest to pass`sd.context` instead of `sd.context.DaemonSetEvictionForEmptyNodes` for latter features.